README.markdown

```diff
@@ -1,144 +1,355 @@
 # puppetlabs-rsync #
 
+#### Table of Contents
+
+1. [Module Description - What does the module do?](#module-description)
+3. [Usage - Configuration options and additional functionality](#usage)
+4. [Reference - An under-the-hood peek at what the module is doing and how](#reference)
+    * [Classes](#classes)
+    * [Defined Types](#defined-types)
+5. [Limitations - OS compatibility, etc.](#limitations)
+
+## Module description
+
 puppetlabs-rsync manages rsync clients, repositories, and servers as well as
 providing defines to easily grab data via rsync.
 
-# Class: rsync #
+## Usage
+
+Install the latest version of rsync
```

Comment:
I *think* you need a line space between the text and the code block on these samples.